### PR TITLE
Remove throw OperationCanceledException in Host.StopAsync

### LINKED ISSUE #40290: Don't throw OperationCanceledException in Host.StopAsync
We shouldn't be throwing in `Host.StopAsync` when the token is canceled and instead let the hosted services and host lifetime run.

We should remove
https://github.com/dotnet/runtime/blob/df7f985f44712efd46b8b3bc5e4d83fe3d83ed4b/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L79
and
https://github.com/dotnet/runtime/blob/df7f985f44712efd46b8b3bc5e4d83fe3d83ed4b/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L91
and add a try catch around
https://github.com/dotnet/runtime/blob/df7f985f44712efd46b8b3bc5e4d83fe3d83ed4b/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L92
and add it to the `AggregateException`

Discussion about this at https://github.com/dotnet/aspnetcore/pull/24279#issuecomment-668208979

cc @Tratcher @halter73

### PR DESCRIPTION
**Pull request by Marusyk**
Removed throw `OperationCanceledExceptio`n in `Host.StopAsync`

Fixes: #40290

Please review
Thank you in advance

### CODE REVIEW

**Comment by Marusyk** [comment]
working on fixing tests

File: src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs [hunk 1]
```diff
@@ -91,9 +90,14 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
...
                 _applicationLifetime.NotifyStopped();
 
-                token.ThrowIfCancellationRequested();
-                await _hostLifetime.StopAsync(token).ConfigureAwait(false);
-
+                try
+                {
+                    await _hostLifetime.StopAsync(token).ConfigureAwait(false);
+                }
+                catch (Exception ex)
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs:97]: Should we catch as `Exception` or a more specific type?

cc: @eerhardt @BrennanConroy

**Review by maryamariyan — APPROVED** [review]
LGTM aside from the one question.

File: src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs (hunk 1 above)
Comment (BrennanConroy) [reply src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs:97]: `Exception` seems fine, hostlifetime can be anything and this is the same as what we do above with hostedservice.

**Comment by maryamariyan** [comment]
The `dotnet-runtime-perf (Performance Linux x64 release wasm wasm micro net5.0)` failure happened again upon retry.

@safern I checked AzDo but not sure if there is a permission issue in the machine. Would be able to check what the problem could be?

**Comment by maryamariyan** [comment]
Seems like the perf runs are no longer relevant and already removed on Friday. If you rebase it should go away.

**Comment by danmosemsft** [comment]
@maryamariyan there is no need to rebase just to remove those dead CI legs. If you are happy with it, you could just merge.

**Review by halter73 — APPROVED** [review]
I'm a bit surprised that I don't see any test verifying Host.StopAsync()'s exception handling behavior in [HostTests](https://github.com/dotnet/runtime/blob/c5b4474a9026f5655b1619541185a6c4c95640fa/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs).

I guess it would be unfair to ask for some tests now, but a regression test would be nice. Particularly one that verifies StopAsync() doesn't throw given a canceled token (assuming  _hostedServices and _hostLifetime also don't throw). That was the regression moving from WebHost to the Generic Host that caused us to notice this.

The change looks good though.

**Comment by halter73** [comment]
I guess this was already merged 😄

**Comment by maryamariyan** [comment]
Thanks @halter73. Opened issue https://github.com/dotnet/runtime/issues/40700 to track your feedback.

**Pull request merged**